kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/check/consistency/algorithm/DataMatchDataConsistencyCalculateAlgorithm.java

```diff
@@ -224,7 +224,20 @@ public boolean equals(final Object o) {
             }
             return true;
         }
-        
+    
+        private boolean checkDifferentNumberTypeMatched(final Number thisResult, final Number thatResult) {
+            if (thisResult instanceof Integer) {
+                return thisResult.intValue() == thatResult.intValue();
+            }
+            if (thisResult instanceof Long) {
+                return thisResult.longValue() == thatResult.longValue();
+            }
+            if (thisResult instanceof Short) {
+                return thisResult.longValue() == thatResult.longValue();
+            }
```

Comment:
1, Why use `intValue` for Integer but `longValue` for other types

2, Does `Byte` need to do convert?